delivery_carrier_label_seur/models/stock.py

```diff
@@ -0,0 +1,262 @@
+# -*- coding: utf-8 -*-
+# © 2015 FactorLibre - Ismael Calvo <ismael.calvo@factorlibre.com>
+# © 2017 PESOL - Angel Moya <angel.moya@pesol.es>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+from odoo.addons.queue_job.job import job
+from datetime import timedelta
+from xml.dom.minidom import parseString
+_logger = logging.getLogger(__name__)
+try:
+    from unidecode import unidecode
+except (ImportError, IOError) as err:
+    _logger.debug(err)
+
+FINAL_SEUR_TRACKING_STATES = [
+    'ENTREGA EFECTUADA',
+    'ENTREGADO EN PUNTO',
+    'ENTREGADO CAMBIO SIN RETORNO',
+]
+
+
+class ShippingLabel(models.Model):
+    _inherit = 'shipping.label'
+
+    @api.model
+    def _get_file_type_selection(self):
+        """ To inherit to add file type """
+        res = super(ShippingLabel, self)._get_file_type_selection()
+        res.append(('txt', 'TXT'))
+        res = list(set(res))
+        return res
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    seur_service_code = fields.Selection(selection='_get_seur_services',
+                                         default=False)
+    seur_product_code = fields.Selection(selection='_get_seur_products',
+                                         default=False)
+    file_type = fields.Selection(
+        related='carrier_id.seur_config_id.file_type', readonly=True)
+
+    def _get_seur_services(self):
+        return self.env['delivery.carrier'].SEUR_SERVICES
+
+    def _get_seur_products(self):
+        return self.env['delivery.carrier'].SEUR_PRODUCTS
+
+    @api.onchange('carrier_id')
+    def carrier_id_change(self):
+        res = super(StockPicking, self).carrier_id_change()
+        if not self.carrier_id or self.carrier_id.carrier_type != 'seur':
+            return
```

Comment:
You are not aware what was done on super.
```suggestion
            return res
```